analogin.py

```diff
@@ -62,19 +59,19 @@ def load_xml(self, node):
             if type(child) == ET.Element:
             
                 if child.tag == "enable":
-                    self.enable = str_to_bool(child.text)
+                    self.enable = Instrument.str_to_bool(child.text)
```

Comment:
Why use Instrument.str_to_bool instead of self if it's not being overwritten?

Reply:
Calling with `Instrument` suggests to me as a reader of the code that the method is static, which provides more information even though `self.foo` is syntactically correct (it wasn't always). It could theoretically also indicate a class method but it is clear from how it is being used that the method is not a class method. It has the same functionality but gives someone reading the code more information.

Reply:
call from self, I'll edit those functions to make use of the self's properties and provide better messages.

Reply:
in particular using self.__class__.__name__ provides a nicer error message

Reply:
Those are utility functions that should be kept static. If we want the `__class__` or `__name__` for error messages then that's why we have loggers characterized by those values. We don't need to re-invent that wheel.

Reply:
It's a simple matter of printing f"...{self.\_\_name__.\_\_class__}..." to make the source of the xml error easier to identify at a glance.
I think the error messages should be as verbose as possible so that mental effort expended by a user is minimal. It'll make debugging much less painful when a small mistake is made by the (understandably tired) user